### ISSUE TITLE
Make it harder to shoot in the foot with %requires_eq

### ISSUE THREAD

**Issue by Vogtinator** [openSUSE/rpm-config-SUSE#27]
docker broke in TW because it switched over to `%requires_eq`, simplified change:
```
BuildRequires: foo = 1.0
-Requires: foo = 1.0
+%requires_eq foo
```

This is for two reasons:
1. `%requires_eq` does not work with capabilities
2. `%requires_eq` fails silently if anything was not found

Fixing 1 is easy, just add `--whatprovides` and adjust the `grep -v`, but I'm not sure how to implement 2.
Arguably, just removing the `grep -v` works well enough, as it results in:
```error: line 11: Unknown tag: no package provides asdf```
though  that seems like a hack. I tried `-e 's/no package provides.*$/%%{error:\0}/', but it appears like that is not expanded recursively.

**Comment by Conan-Kudo** [comment]
I dunno, I could just drop it, that eliminates a huge class of footguns...

**Comment by DimStar77** [comment]
The issue with ```--what-provides``` is that this has the potential of returning more than one package name

**Comment by Vogtinator** [comment]
And also a huge class of successful builds... The macro is still useful, dropping it would only lead to having badly copied versions everywhere.

**Comment by Vogtinator** [comment]
> The issue with --whatprovides is that this has the potential of returning more than one package name

Right, which makes `%requires_eq` quite ambiguous in those cases. So it should probably error out in those cases too.

**Comment by DimStar77** [comment]
> 
>     2. `%requires_eq` fails silently if anything was not found

https://github.com/openSUSE/rpm-config-SUSE/pull/64 would address that
